Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -289,7 +289,35 @@ Further brute force mitigation techniques are described on the OWASP page [Block
 
 #### Dynamic Analysis
 
-Attempt to log in with an incorrect password multiple times. After three to five attempts, your login should be (at least temporarily) rejected even when correct credentials are entered.
+The purpose of dynamically analysing authentication mechanisms is to make sure an important number of login attempts cannot be made to guess credentials. 
+
+To test the level of confidence the authentication mechanism provides in an app, automation is often the preferred way. Many tools can be used, including scripts and proxies; the choice for the relevant tool may differ depending upon the situation. 
+For this demo, we'll use a proxy (Burp Suite Free edition). Burp Suite offers a module dedicated to such attacks called 'Intruder' (more information can be found from Burp Suite official documentation at [Burp Suite official documentation](https://portswigger.net/burp/help/ "Burp Suite official documentation")):
+
+  - start Burp Suite;
+  - create a new project (or open an existing one);
+  - enter relevant parameters for the project;
+  - now that the proxy is set between your app and the targeted web site, browse the site and intercept the request that corresponds to the authentication mechanism;
+  - in the 'Proxy / HTTP History' tab, right-click this request and select 'Send to Intruder' in the list; 
+  - select the 'Intruder' tab in Burp Suite;
+  - make sure all parameters in the 'Target', 'Positions' and 'Options' tabs are set accordingly, and select the 'Payload' tab;
+  - load the list of passwords to be tried or upload the list. The attack is ready to be started!
+  
+![List of passwords in Burp Suite](Images/Chapters/0x04e/BurpIntruderInputList.gif)
+  
+  - click the 'Start attack' button to launch the attack on the authentication mechanism.
+  
+A new window opens and requests to the site are sent in sequence, each request corresponding to a password in the provided list. For each request, information on the response is provided (length, status code, ...), which allows to distinguish the successful attempt from the others :
+
+![A successful attack in Burp Suite](Images/Chapters/0x04e/BurpIntruderSuccessfulAttack.gif)
+
+In this example, the successful attempt can be identified from its different length (password = "P@ssword1"). 
+
+In order to prevent this type of attack from being successful, a few mitigations can be put in place :
+  - after a few unsuccessful attempts, targeted accounts should be temporarily locked and any kind of attempt (successful or unsuccessful) should be rejected in the same manner.
+  - change default credentials and avoid common usernames and passwords.
+  - put a strong emphasis on protection privileged accounts.
```

Comment:
Not necessarily, I mean it's more worrying to have privileged accounts hacked than regular accounts.
I think the sentence is not clear, I will rephrase.